tests/python/test_lists.py

```diff
@@ -0,0 +1,977 @@
+    a = pm.eval("[2]")
+    a.reverse()
+    assert a == [2]  
+
+def test_reverse_too_many_args():
+    a = pm.eval('([1,2,3,4])')
+    try:
+        a.reverse(3)     
+        assert (False)
+    except Exception as e:    
+        assert str(type(e)) == "<class 'TypeError'>"
+        assert str(e) == "JSArrayProxy.reverse() takes no arguments (1 given)"     
+
+#sort
+def test_sort():
+    a = pm.eval("[5,1,2,4,9,6,3,7,8]")
+    a.sort()
+    assert a == [1,2,3,4,5,6,7,8,9]       
+
+def test_sort_reverse():
+    a = pm.eval("[5,1,2,4,9,6,3,7,8]")
+    a.sort(reverse=True)
+    assert a == [9,8,7,6,5,4,3,2,1]   
+
+def test_sort_reverse_false():
+    a = pm.eval("[5,1,2,4,9,6,3,7,8]")
+    a.sort(reverse=False)
+    assert a == [1,2,3,4,5,6,7,8,9]      
+
+def test_sort_with_function():
+    def myFunc(e,f):
+        return len(e) - len(f)     
+    a = pm.eval("(['Ford', 'Mitsubishi', 'BMW', 'VW'])")
+    a.sort(key=myFunc)
+    assert a == ['VW', 'BMW', 'Ford', 'Mitsubishi']   
+
+def test_sort_with_js_function():
+    a = pm.eval("(['Ford', 'Mitsubishi', 'BMW', 'VW'])")
+    myFunc = pm.eval("((a, b) => a.toLocaleUpperCase() < b.toLocaleUpperCase() ? -1 : 1)")
+    a.sort(key=myFunc)
+    assert a == ['BMW', 'Ford', 'Mitsubishi', 'VW']    
+
+# TODO does not report exception for one-arg function
+#def test_sort_with_one_arg_function():
+#    def myFunc(e):
+#        return len(e)     
+#    a = pm.eval("(['Ford', 'Mitsubishi', 'BMW', 'VW'])")
+#    a.sort(key=myFunc)
+#    assert a == ['VW', 'BMW', 'Ford', 'Mitsubishi']  
+
+def test_sort_with_function_and_reverse_false():
+    def myFunc(e,f):
+        return len(e) - len(f)     
+    a = pm.eval("(['Ford', 'Mitsubishi', 'BMW', 'VW'])")
+    a.sort(key=myFunc, reverse=False)
+    assert a == ['VW', 'BMW', 'Ford', 'Mitsubishi']   
+
+def test_sort_with_js_function_and_reverse_false():
+    a = pm.eval("(['Ford', 'Mitsubishi', 'BMW', 'VW'])")
+    myFunc = pm.eval("((a, b) => a.toLocaleUpperCase() < b.toLocaleUpperCase() ? -1 : 1)")
+    a.sort(key=myFunc, reverse=False)
+    assert a == ['BMW', 'Ford', 'Mitsubishi', 'VW']    
+
+def test_sort_with_function_and_reverse():
+    def myFunc(e,f):
+        return len(e) - len(f)     
+    a = pm.eval("(['Ford', 'Mitsubishi', 'BMW', 'VW'])")
+    a.sort(key=myFunc, reverse=True)
+    assert a == ['Mitsubishi', 'Ford', 'BMW', 'VW']
+
+def test_sort_with_js_function_and_reverse():
+    a = pm.eval("(['Ford', 'Mitsubishi', 'BMW', 'VW'])")
+    myFunc = pm.eval("((a, b) => a.toLocaleUpperCase() < b.toLocaleUpperCase() ? -1 : 1)")
+    a.sort(key=myFunc, reverse=True)
+    assert a == ['VW', 'Mitsubishi', 'Ford', 'BMW']      
+
+def test_sort_with_function_wrong_type():
+    a = pm.eval('([1,2,3,4])')
+    try:
+        b = 9
+        a.sort(key=b)    
+        assert (False)
+    except Exception as e:    
+        assert str(type(e)) == "<class 'TypeError'>"
+        assert str(e) == "'int' object is not callable"       
```

Comment:
The `key` function supplied to `list.sort()` should only take 1 argument, and can return any value. See my comment on JSArrayProxy_sort for more details.